### PR TITLE
RATEST-92:Failing Test: Contact Info Test

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-92  cc @dkayiwa @ibacher  @kdaud 

Opened a new clean Pr
Removed a line  which was showing an assert method that made  the server to take long,it was impossible for the function to add more waiting capabilities for the driver to listen to the assert, so i removed it from the functionality, i would have improved the assertions but the assert at the end of the line causes the server not to quite hence throwing exceptions

### CODE REVIEW

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:69]: We need the `assertTrue()` method in this test to assert that the recorded telephone number by the user is stored in the `db`  and that is why this test is current failing simply because by the time it confirms whether the telephone number is present it finds a `null value` and the test fails. So by removing the assertion makes the test meaningless(`tasteless`).

As mentioned earlier the pre-written test logic workflow is not not correct. The leakage is at this line number [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/9a7dc58fae262f0444891e3bc4af59cedf31867b/ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java#L55) that messes up things and by the time the test reaches the `assertTrue()` function which you have proposed to git rid of, the test  finds no telephone number that was recorded.

FYI for the past we had tests that almost had almost a similar issue and this [ticket ](https://issues.openmrs.org/browse/RATEST-75) was created not cater for the tests mystery and the committer to that ticket left a comprehensive report that might be a good one to take a look.

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:69]: You ,are right by saying we need assertTrue ,However i really dont agree with your suggestion about the link where this test is breaking, this test was breaking from [here](https://github.com/openmrs/openmrs-contrib-uitestframework/pull/45). so ideally this test is in the right condition , the only improvement is to fix the assert

**Comment by sherrif10** [comment]
Hey @dkayiwa  @ibacher , this assertTrue(patientDashboardPage.getTelephoneNumber().contains("111111111")); seems to be breaking this test and i think the only problem is contains method takes in single argument which is a string by default, on patientdashboard we only find 11111 not "1111" thats why the driver is failing to see it, could you be having a suggestion

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (ibacher) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:69]: Have we considered the possibility that `patientDashboardPage.getTelephoneNumber()` needs to be fixed? The [selector here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/2b9f429c66f23f636c27ea2b14e20e34f74a1884/ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java#L79) doesn't select anything for me and could be simplified to `#coreapps-telephoneNumber`, which is the ID of the field containing the telephone number.

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (ibacher) [inline ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:67]: I'm a little confused as to why we need to do this here if there are no more asserts after it. In general, things to reset the test to the appropriate state should be run in `@After` commands. This ensures that they are run even if the test fails (otherwise a single test failure can cause a cascade of failures).

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:67]: Thanks @ibacher , Actually this test is really fine, we would have just removed @ignore annotation but now its failing on the asserts, i really aggree with adding assert in @After method, though still assert is wrong Let me try that and see

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:69]: ohh thats a nice catch , let me fix that quickly

**Comment by kdaud** [comment]
@ibacher as mentioned [here](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/522#discussion_r683990637) the test has a leakage that is hidden and the test comes break at a different point far a way from it that can make one think there can not exist a leakage. I as well gave a similar scenario with references that we had on some of those tests that have been resurrected. I pointed out where the mess is located. Not until the leakage is repaired then the test will pass!

**Comment by sherrif10** [comment]
Thanks @ibacher @kdaud  , @ibacher  True getTelephoneNumber had issues with the element, i collected it but am falling into indexoutOfboundExceptions with in the method they logacally created here   https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java#L397  which i will fix as soon as possible

File: ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java
Comment (ibacher) [inline ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java:400]: I'm really confused about what this is supposed to do? Why do we have two variables? Where are `i` and `j` used?

Comment (ibacher) [reply ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java:400]:
```suggestion
            phoneNumber = phoneNumber.trim();
```

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java:400]: Sorry ,Will make it more meaningfull.

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java]: Thanks @ibacher i had messed up recently

**Comment by sherrif10** [comment]
cc @ibacher  @dkayiwa

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:40]: Not sure why you added the empty line `LN-40` which was not there before. To have consistency within our code base, I would recommend you omit the line. Cross check other existing test methods within the other test and find out whether they do have this empty.

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:69]: Once we have the presence of the telephone number  in the db asserted, does the test need to go back to home page? I think once the test asserts the number then it should then go to teardown() function.

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:70]: Do we really need line 70 which was not there before?

**Comment by sherrif10** [comment]
cc @dkayiwa  @ibacher  @kdaud  Looking forward for your reviews thanks

**Comment by dkayiwa** [comment]
@sherrif10 can you reduce the number of times you ping us to only when @kdaud has reviewed and approved?

**Comment by sherrif10** [comment]
ohk thanks @dkayiwa.  cc @kdaud .

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java]: Resolved thanks

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:69]: True we can let it finish from assert. Thanks.

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java:40]: For code quality and readability we might need this line since we have been ephasizing this clean code convention style

File: ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java]: Resolved thanks.

**Comment by sherrif10** [comment]
Hello @kdaud  waiting for your review thanks

**Comment by kdaud** [comment]
@sherrif10 have you simulated the test on all `instances` and the changes are passing the test ?

**Comment by sherrif10** [comment]
The tests are passing very well on all instances , qa, demo , local instance

**Comment by kdaud** [comment]
Hear is my experience. 
When I simulate the test with the suggested changes, the test is breaking at the line code [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/9dfa946b71ffdc1e8aafd29e70394fe5c4689834/ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java#L53) on either instances. Not sure why ? 
Wouldn't be a good idea to take a look into it ?

**Comment by sherrif10** [comment]
> the test is breaking at the line code [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/9dfa946b71ffdc1e8aafd29e70394fe5c4689834/ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java#L53) on either instance

Have never fall into that line breaking this issue before so really i cant tell exactly why its breaking on that side, it would have been broken here as well,May be try to apply a debug point on that line and confirm.

**Comment by sherrif10** [comment]
Hello @kdaud ,Any more suggestions or approval

**Comment by kdaud** [comment]
> ,Any more suggestions or approval

Am taking a simulation locally once again with the suggested changes and will feedback you soon otherwise great work @sherrif10

**Comment by sherrif10** [comment]
Before you take a simulation, Make sure your uitestframework is update build it with mvn clean install to pull all the artifact dependenciess , otherwise you will still see a failure in this test.

**Comment by kdaud** [comment]
Here is the lab report after simulation!
The test is passing well on all instances!

Few things that might be worth looking into before the test enters into master
Remove the redundant line [28](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/9dfa946b71ffdc1e8aafd29e70394fe5c4689834/ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java#L28) for consistency within our code base.

Visit the code changes [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/9dfa946b71ffdc1e8aafd29e70394fe5c4689834/ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java#L26) and pay a revisit at the conversation [here](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/450#discussion_r681447476) and more additional resource [here](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/450#discussion_r681587355)

**Comment by sherrif10** [comment]
> Remove the redundant line [28](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/9dfa946b71ffdc1e8aafd29e70394fe5c4689834/ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java#L28) for consistency within our code base.

i guess that line is not redandunt however it actually makes it more clear to the reader to have clear difference than the next line.

**Comment by sherrif10** [comment]
Resolved the server failures. cc @kdaud

**Comment by kdaud** [comment]
@sherrif10 did you take a look at the last paragraph at the review [above](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/522#issuecomment-896686876) ?

**Comment by sherrif10** [comment]
> did you take a look at the last paragraph in the review [above](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/522#issuecomment-896686876) ?

Is there something missing with this commit .

**Comment by kdaud** [comment]
@sherrif10 have you taken the initiative to revisit the last paragraph in the review [above](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/522#issuecomment-896686876) ?

**Comment by sherrif10** [comment]
Resolved thanks.

File: ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: @sherrif10 what do you think about getting a string name more readable than `STR` ?

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java]: Nice catch. thanks Resolved

File: ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: @sherrif10 what is your thought about `PHONENUMBER_VALIDATOR` vs `PHONE_NUMBER_VALIDATOR` vs `TELEPHONE_VALIDATOR` ? Which one would be the best for you?

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: > PHONE_NUMBER_VALIDATOR

Sure this sounds good

**Review by kdaud — APPROVED** [review]
Welcoming `ContactInforTest` in the world of master builds. Thanks @sherrif10 for redeeming the test from @ignored-world !

Waiting either @dkayiwa or @ibacher to do the magic of merging and see the test inclusive in master builds!

File: ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java
Comment (dkayiwa) [inline ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: Is this the actual validator or validation message?

Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: > Is this the actual validator or validation message?

What a question!

Waiting to hear @sherrif10 's response!

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: > Is this the actual validator or validation message?

Its a validation message. Thanks @dkayiwa

Comment (dkayiwa) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: I thought the answer would be a followup commit. 😊

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: @dkayiwa , Didnt get that clearly .

Comment (dkayiwa) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: Let me give you some time to think through my original message and see if you have to do something. 😊

Comment (dkayiwa) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java]: `NOTIFICATION` is confusing.

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java]: Resolved with _PHONE_NUMBER_MESSAGE_ . How about phone_number_message.

File: ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java
Comment (dkayiwa) [inline ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java:26]: PHONE_NUMBER_VALIDATION_MESSAGE would be clearer

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/RegistrationEditSectionPage.java]: Thanks @dkayiwa  For reviewing and guidance. Resolved.

**Comment by kdaud** [comment]
Now that @sherrif10 you have resolved [ContactInfoTest](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java) would you like to make a follow up on [RATEST-79](https://issues.openmrs.org/browse/RATEST-79) to support the current `assignee `to be able resolve it having got the experience of `ContactInfoTest` since they are almost sister tests in its logic implementation ?

I can see the assignee to that ticket have shared above has taken months figuring out the test and has passion to seeing that test resolved. 

I would love to hear @sherrif10's suggestion on the opinion above?
cc: @dkayiwa @ibacher

**Comment by kdaud** [comment]
> Let me follow up with the ticket

Great for taking the initiative!

@sherrif10 you can as well see more of her insight/findings on the test via the public talk post [here](https://talk.openmrs.org/t/xssonphonenumberfieldtest-tests-are-really-failing/34036).

**Pull request merged**
